src/utils/alert.ts

```diff
@@ -145,49 +151,81 @@ export const sendAlerts = async ({
               ),
             },
             notification
-          ).then(() => ({
+          )
+          return {
             notification: 'mailgun',
             alert: validation.alert,
             url,
-          }))
+          }
         }
         case 'webhook': {
-          return sendWebhook({
+          await sendWebhook({
             ...notification.data,
             body: {
               url,
               alert: validation.alert,
               time: new Date().toLocaleString(),
             },
-          } as WebhookData).then(() => ({
+          } as WebhookData)
+          return {
             notification: 'webhook',
             alert: validation.alert,
             url,
-          }))
+          }
         }
         case 'slack': {
-          return sendSlack({
+          await sendSlack({
             ...notification.data,
             body: {
               url,
               alert: validation.alert,
               time: new Date().toLocaleString(),
             },
-          } as WebhookData).then(() => ({
+          } as WebhookData)
+          return {
             notification: 'slack',
             alert: validation.alert,
             url,
-          }))
+          }
         }
         case 'smtp': {
           const transporter = createSmtpTransport(notification.data as SMTPData)
-          return sendSmtpMail(transporter, {
+          await sendSmtpMail(transporter, {
             // TODO: Read from ENV Variables
             from: 'http-probe@hyperjump.tech',
             to: (notification?.data as SMTPData)?.recipients?.join(','),
             subject: message.subject,
             html: message.body,
           })
+
+          return Promise.resolve({
+            notification: 'smtp',
+            alert: validation.alert,
+            url,
+          })
+        }
+        case 'whatsapp': {
+          const data = notification.data as WhatsappData
+          const token = await loginUser(data)
+
+          if (token) {
+            await Promise.all(
+              data.recipients.map(async (recipient) =>
+                sendTextMessage({
+                  recipient,
+                  token,
+                  baseUrl: data.url,
+                  message: validation.alert,
+                })
```

Comment:
Please refactor this. Create a function that can accept multiple recipients, or change so that `sendTextMessage` accepts array of recipients instead. Then we don't need to map the `data.recipients` here.

Reply:
fixed